src/ifcopenshell-python/ifcopenshell/file.py

```diff
@@ -86,42 +101,75 @@ def __getitem__(self, key):
+        """Return IFC objects filtered by IFC ID.
 
-        Returned objects are unwrapped IFC objects.
+        :param id: STEP numerical identifier
+        :type id: int
+        :returns: Returned objects are unwrapped IFC objects.
+        :rtype: ifcopenshell.entity_instance.entity_instance
         """
         return self[id]
 
     def by_guid(self, guid):
-        """
-        Return IFC objects filtered by IFC GUID.
+        """Return IFC objects filtered by IFC GUID.
 
-        Returned objects are unwrapped IFC objects.
+        :param guid: GlobalId value in 22-character encoded form
+        :type guid: string
+        :returns: Returned objects are unwrapped IFC objects.
+        :rtype: ifcopenshell.entity_instance.entity_instance
         """
         return self[guid]
 
     def add(self, inst):
+        """Undocumented function"""
         inst.wrapped_data.this.disown()
         return entity_instance(self.wrapped_data.add(inst.wrapped_data))
 
     def by_type(self, type):
-        """
-        Return IFC objects filtered by IFC Type and wrapped with the entity_instance class.
+        """Return IFC objects filtered by IFC Type and wrapped with the entity_instance class.
 
-        See ifcopenshell.entity_instance for class methods and properties
+        If an IFC type class has subclasses, all entities of those subclasses are also returned.
+
+        :param type: The case insensitive type of IFC class to return.
+        :type type: string
+        :returns: A list of ifcopenshell.entity_instance.entity_instance objects
+        :rtype: list
         """
         return [entity_instance(e) for e in self.wrapped_data.by_type(type)]
 
     def traverse(self, inst, max_levels=None):
+        """Get a list of all referenced entities for a particular entity
+
+        :param inst: The entity instance to get all sub instances
+        :type inst: ifcopenshell.entity_instance.entity_instance
+        :param max_levels: How far deep to recursively fetch sub instances. None or -1 means infinite.
+        :type max_levels: None|int
+        :returns: A list of ifcopenshell.entity_instance.entity_instance objects
+        :rtype: list
+        """
         if max_levels is None:
             max_levels = -1
         return [entity_instance(e) for e in self.wrapped_data.traverse(inst.wrapped_data, max_levels)]
 
     def get_inverse(self, inst):
+        """Return a list of entities that reference this entity
+
+        :param inst: The entity instance to get inverse relationships
+        :type inst: ifcopenshell.entity_instance.entity_instance
+        :returns: A list of ifcopenshell.entity_instance.entity_instance objects
+        :rtype: list
+        """
         return [entity_instance(e) for e in self.wrapped_data.get_inverse(inst.wrapped_data)]
 
     def remove(self, inst):
+        """Deletes an IFC object in the file.
+
+        Other entities that reference the deleted object will be set to null.
```

Comment:
I would say *Attribute values in other entity instances that ...*. Also in case of a list/set of references, the reference to the deleted will simply get removed from the aggregate.

Reply:
Fixed